addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/handler/AirVisualNodeHandler.java

```diff
@@ -0,0 +1,271 @@
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.airvisualnode.internal.config.AirVisualNodeConfig;
+import org.openhab.binding.airvisualnode.internal.json.NodeData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import jcifs.smb.NtlmPasswordAuthentication;
+import jcifs.smb.SmbFile;
+import jcifs.smb.SmbFileInputStream;
+
+/**
+ * The {@link AirVisualNodeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Victor Antonovich - Initial contribution
+ */
+public class AirVisualNodeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AirVisualNodeHandler.class);
+
+    public static final String NODE_JSON_FILE = "latest_config_measurements.json";
+
+    private final Gson gson;
+
+    private ScheduledFuture<?> pollFuture;
+
+    private long refreshInterval;
+
+    private String nodeAddress;
+
+    private String nodeUsername;
+
+    private String nodePassword;
+
+    private String nodeShareName;
+
+    private NodeData nodeData;
+
+    public AirVisualNodeHandler(@NonNull Thing thing) {
+        super(thing);
+        gson = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing AirVisual Node handler");
+
+        AirVisualNodeConfig config = getConfigAs(AirVisualNodeConfig.class);
+
+        if (config.address == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Node address must be set");
+        }
+        this.nodeAddress = config.address;
+
+        this.nodeUsername = config.username;
+
+        if (config.password == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Node password must be set");
+        }
+        this.nodePassword = config.password;
+
+        this.nodeShareName = config.share;
+
+        this.refreshInterval = config.refresh * 1000L;
+
+        schedulePoll();
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Node poll scheduled");
```

Comment:
This will always hide the 2 previous configuration errors.

Reply:
since you added the `return` statements above you will always end up with an offline thing which in reality is still in `INITIALIZING` state, I would suggest to remove this line and set the thing to ONLINE or OFFLINE inside your polling job.